src/gui/base/Overlay.ts

```diff
@@ -95,20 +104,32 @@ export const overlay: Component = {
 							height: position.height,
 							"z-index": position.zIndex != null ? position.zIndex : LayerType.Overlay,
 						},
-						oncreate: (vnode) => {
-							const dom = vnode.dom as HTMLElement
-							overlayAttrs[1] = dom
-
-							if (attrs.createAnimation) {
-								animations.add(dom, attrs.createAnimation(dom))
+						onanimationend: () => {
+							if (attrs.isClosing) {
+								overlays.delete(key)
 							}
 						},
-						onremove: () => {
-							overlayAttrs[1] = null
+						onupdate(vnode: VnodeDOM<any>): any {
+							if (attrs.isClosing && attrs.closeAnimation != null) {
+								const dom = vnode.dom as HTMLElement
+
+								// Force the environment to restart the animations via a reflow
```

Comment:
not only this is hacky but also causes a relayout?

Reply:
It is hacky but it seems to be the standard technique afaik. The 'relayout' (the term MDN & Google use is 'reflow') is deliberate as animations are only played on reflow.

Reply:
Unfortunately their workaround is not applicable & the reflow is still needed but it did help simplify the code.